Build/cake/unit-tests.cake

```diff
@@ -0,0 +1,34 @@
+#tool "nuget:?package=Microsoft.TestPlatform&version=15.7.0"
+#tool "nuget:?package=NUnitTestAdapter&version=2.1.1"
+
+Task("EnsureAllProjectsBuilt")
+  .IsDependentOn("CompileSource")
```

Comment:
We can remove the dependent on `CompileSource` step since we are calling MSBuild manually inside of this step?

Reply:
Doesn't have to be `CompileSource` but we'd need at least 

```suggestion
  .IsDependentOn("UpdateDnnManifests")
  .IsDependentOn("Restore-NuGet-Packages")
```

to ensure that the unit tests target can be independently executed in a clean workspace

Reply:
That sounds better to me, I do not like the idea of compiling twice.